src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.cs

```diff
@@ -2190,7 +2206,7 @@ private XClassName GetClassName(XamlObjectDefinition control)
 
 				var index = fullName.LastIndexOf('.');
 
-				return new(fullName.Substring(0, index), fullName.Substring(index + 1), FindType(fullName));
+				return new(fullName.Substring(0, index), fullName.Substring(index + 1), _metadataHelper.FindTypeByFullName(fullName) as INamedTypeSymbol);
```

Comment:
In the work for StringBuilder, maybe we can avoid the multiple substrings with StringBuilder manipulations ?

Reply:
@jeromelaban The constructor here takes two strings, one for namespace and one for class name. I'm not sure how we can benefit from StringBuilder.

If the `Namespace` and `ClassName` are not always accessed, we could benefit from making their evaluation lazy.